appveyor_before_build.ps1

```diff
@@ -0,0 +1,40 @@
+. .\appveyor_helpers.ps1
+
+#
+# Copies the C# plugin so that the SonarQube MSBuild Scanner packaging projects will patch it. The packaging project is invoked by Appveyor as part of the regular build.
+#
+function CopyCsharpPluginForPatching
+{
+    Add-AppveyorMessage -Message "Copying the C# plugin for patching"
+        
+    $csPluginCleanJar = FindSingleFile "$mavenLocalRepository\org\sonarsource\dotnet\sonar-csharp-plugin" "*SNAPSHOT.jar"
```

Comment:
imo it'd be cleaner here not to rely on the Maven repo: Your'e checking out the sources of the C# plugin, and so you should know the location of the jar (`target\sonar-csharp-plugin-sources.jar`).

then you can also run `mvn clean package` instead of `install`, which will produce the .jar but not install it in the local Maven repo.